src/Controls/src/Core/Handlers/Items/Android/RecyclerViewScrollListener.cs

```diff
@@ -55,23 +55,42 @@ public override void OnScrolled(RecyclerView recyclerView, int dx, int dy)
 
 			_itemsView.SendScrolled(itemsViewScrolledEventArgs);
 
-			// Don't send RemainingItemsThresholdReached event for non-linear layout managers
-			// This can also happen if a layout pass has not happened yet
-			if (Last == -1)
+			// Early returns for edge cases
+			if (Last == -1 || ItemsViewAdapter == null || _itemsView.RemainingItemsThreshold == -1)
+			{
+				return;
+			}
+
+			var itemsSourceCount = ItemsViewAdapter.ItemCount;
+			bool hasHeader = ItemsViewAdapter.ItemsSource.HasHeader;
+			bool hasFooter = ItemsViewAdapter.ItemsSource.HasFooter;
+
+			int firstDataItemIndex = (hasHeader && hasFooter) ? 1 : 0;
```

Comment:
With the changes, `firstDataItemIndex` is set to 1 only when both Header and Footer exist. However, if only Header is present, the value should be 1 too, right?

Reply:
Yes @jsuarezruiz , `firstDataItemIndex` should be set to `1` whenever only the header is present. Since the header and footer affect indexing independently, they should be evaluated separately. 
 
I've updated the logic to reflect this and optimized the code for improved clarity and efficiency. Please review the modified fix and let me know if any further changes are required.